docs/develop/java/spring-boot-integration.mdx

```diff
@@ -0,0 +1,250 @@
+  workers-auto-discovery:
+    packages:
+      - your.package # enumerate all the packages that contain your workflow implementations.
+```
+
+#### What is auto-discovered:
+
+Workflows implementation classes annotated with `io.temporal.spring.boot.WorkflowImpl`
+Activity beans present Spring context whose implementations are annotated with `io.temporal.spring.boot.ActivityImpl`
+Nexus Service beans present in Spring context whose implementations are annotated with `io.temporal.spring.boot.NexusServiceImpl`
+Workers if a Task Queue is referenced by the annotations but not explicitly configured. Default configuration will be used.
+
+:::note
+`io.temporal.spring.boot.ActivityImpl` and `io.temporal.spring.boot.NexusServiceImpl` should be applied to beans, one way to do this is to annotate your Activity implementation class with `@Component`
+:::
+
+```
+@Component
+@ActivityImpl(workers = "myWorker")
+public class MyActivityImpl implements MyActivity {
+  @Override
+  public String execute(String input) {
+    return input;
+  }
+}
+```
+
+:::note
+Auto-discovered Workflow implementation classes, Activity beans, and Nexus Service beans will be registered with the configured Workers if not already registered.
+:::
+
+## Interceptors {#interceptors}
+
+To enable Interceptors, users can create beans implementing the `io.temporal.common.interceptors.WorkflowClientInterceptor`, `io.temporal.common.interceptors.ScheduleClientInterceptor`, or `io.temporal.common.interceptors.WorkerInterceptor` interface. Interceptors will be registered in the order specified by the `@Order` annotation.
+
+## Integrations {#integrations}
+
+The Temporal Spring Boot integration also has built in support for various tools in the Spring ecosystem, such as metrics and tracing.
+
+#### Metrics
+
+You can set up built-in Spring Boot Metrics using [Spring Boot Actuator](https://docs.spring.io/spring-boot/reference/actuator/metrics.html). The Temporal Spring Boot integration will pick up the MeterRegistry bean and use it to report Temporal Metrics.
+
+Alternatively, you can define a custom `io.micrometer.core.instrument.MeterRegistry` bean in the application context.
+
+#### Tracing
+
+You can set up [Spring Cloud Sleuth](https://spring.io/projects/spring-cloud-sleuth) with OpenTelemetry export. The Temporal Spring Boot integration will pick up the OpenTelemetry bean configured by spring-cloud-sleuth-otel-autoconfigure and use it for Temporal Traces.
+
+Alternatively, you can define a custom `io.opentelemetry.api.OpenTelemetry` for OpenTelemetry or `io.opentracing.Tracer` for OpenTracing bean in the application context.
+
+## Customization of Options {#customize-options}
+
+To programmatically customize the various options that are created by the Spring Boot integration, users can create beans that implement the `io.temporal.spring.boot.TemporalOptionsCustomizer<OptionsBuilderType>` interface. This will be called after the options in your properties files are applied.
```

Comment:
```suggestion
To programmatically customize the various options that are created by the Spring Boot integration, you can create beans that implement the `io.temporal.spring.boot.TemporalOptionsCustomizer<OptionsBuilderType>` interface. This will be called after the options in your properties files are applied.
```